debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java

```diff
@@ -232,7 +256,7 @@ private void determineCapturedTables(SnapshotContext ctx) throws Exception {
             }
         }
 
-        ctx.capturedTables = capturedTables;
+        ctx.capturedTables = sort(capturedTables);
```

Comment:
What if one pattern matches more than one table? Would the order be deterministic? Maybe the captured tables should be also sorted to ensure exact order for such case.

Reply:
I am sorry I do not fully understand what you mean. I had an Integration test with some overlapping patterns. Do you think it is enough and answer your question ?

Reply:
If you will have a single regex that will capute tables aa, ab, ac - could you predict in which order they'll be captured?

Reply:
Actually no. Do you think we should sort by String "natural order" ? For me it is not really relevant but maybe there are some use cases that I do not see right now.

Reply:
I could just sort the stream like that :
````
private Stream<TableId> toTableIds(Set<TableId> tableIds, Pattern pattern) {
        return tableIds
                .stream()
                .filter(tid -> pattern.asPredicate().test(tid.toString()))
                .sorted();
    }
````

Reply:
PR updated.

Reply:
PR updated : capturesTables is sorted by natural order (even if table.whitelist is missing).